src/xref2/env.ml

```diff
@@ -631,17 +631,26 @@ let rec open_signature : Odoc_model.Lang.Signature.t -> t -> t =
             let ty = Component.Delayed.put (fun () -> module_ empty t) in
             add_module
               (t.Odoc_model.Lang.Module.id :> Identifier.Path.Module.t)
-              ty (docs empty t.L.Module.doc) env
+              ty
+              (docs empty t.L.Module.doc)
+              env
         | Odoc_model.Lang.Signature.ModuleType t ->
             let ty = module_type empty t in
             add_module_type t.Odoc_model.Lang.ModuleType.id ty env
         | Odoc_model.Lang.Signature.Comment c -> add_comment c env
         | Odoc_model.Lang.Signature.TypExt te ->
+            let parent =
+              match te.L.Extension.constructors with
+              | [] -> assert false
+              | { id = `Extension (parent, _); _ } :: _ -> parent
+            in
```

Comment:
Parent should be the current signature. Unfortunately, the identifier is not available in this scope.

Reply:
We could add the parent to `Lang.Extension.t` as we have with `Include.t`